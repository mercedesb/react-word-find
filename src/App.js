import React, { Component } from 'react';
import './App.css';
import { WordBank } from './WordBank'
import { PuzzleGenerator } from './PuzzleGenerator'

class App extends Component {
  state = {
    words: ['hello', 'world'],
    shouldGenerate: false
  }

  onAdd = (word) => {
    const currentWordBank = this.state.words
    currentWordBank.push(word)
    this.setState({words: currentWordBank})
  }

  generateWordFind = () => {
    this.setState({shouldGenerate: true})
  }

  buildPuzzle = () => {
    if (this.state.shouldGenerate) {
      return (
        <PuzzleGenerator
          wordBank={this.state.words}
        />
      )
    }
  }

  render () {
    return (
      <div className='App'>
        <WordBank
          words={this.state.words}
          onAdd={this.onAdd}
        />
        <button className='Button Button--ghost' onClick={this.generateWordFind}>Generate Word Find</button>
        {
          this.buildPuzzle()
        } 
      </div>
    )
  }
}

export default App;

