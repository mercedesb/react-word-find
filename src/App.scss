@import url("https://fonts.googleapis.com/css?family=Proxima+Nova");

$base-spacing: 10px;
$brand-color: #44957c;
$font-family: "Proxima Nova", Arial, sans-serif;
$white: #FFFFFF;
$black: #636668;

html {
  height: 100%;
}

body, #root {
  min-height:100%;
  display: flex;
  flex-direction: column;
  flex: 1;
}

.App {
  flex: 1;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $brand-color;
  font-family: $font-family;
  color: $white;
  min-height:100%;
}

.Button {
  background-color: $brand-color;
  border: 0;
  border-radius: 3px;
  color: $white;
  cursor: pointer;
  display: inline-block;
  font-family: $font-family;
  font-size: 1.2rem;
  font-weight: 600;
  line-height: 1;
  padding: 1em 2em;
  margin-bottom: 1em;
  text-decoration: none;
  transition: background-color 150ms ease;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;

  &--ghost {
    background-color: transparent;
    border: 1px solid $white;

    &:hover {
      background-color: rgba(255, 255, 255, 0.1)
    }
  }
}

input {
  background-color: $white;
  color: $black;
  border: 1px solid $white;
  border-radius: 3px;
  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.06);
  box-sizing: border-box;
  font-family: $font-family;
  font-size: 1.2rem;
  margin: 0 1em 1em 0;
  padding: 1em;
  transition: border-color 150ms ease;
  /*width: 100%;*/
}

.WordBank {
  margin: 2em;
  width: 700px;

  &-container {
    background-color: $white;
    color: $black;
    padding: 2em;
    display: flex;
    flex-wrap: wrap;
  }

  &-word {
    font-size: 1.6em;
    padding: 1em;
  }

  &-addContainer {
    display: flex;
    margin: 1em 0;
  }
}

.WordPuzzle {
  display: grid;
  background-color: $white;
  color: $black;
  padding: 2em;
  margin: 2em;
  font-size: 2em;

  &--seven {
    grid-template-rows: repeat(7, auto);
    grid-template-columns: repeat(7, 40px);
  }

  &-letter {
    padding: 0 $base-spacing;
  }
}