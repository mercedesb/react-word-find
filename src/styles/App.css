@import url("https://fonts.googleapis.com/css?family=Proxima+Nova");
html {
  height: 100%; }

body, #root {
  min-height: 100%;
  display: flex;
  flex-direction: column;
  flex: 1;
  align-items: center;
  background-color: #44957c; }

.App {
  flex: 1;
  text-align: center;
  display: flex;
  flex-direction: column;
  background-color: #44957c;
  font-family: "Proxima Nova", Arial, sans-serif;
  color: #FFFFFF;
  min-height: 100%;
  max-width: 1000px; }

.Button {
  background-color: #44957c;
  border: 0;
  border-radius: 3px;
  color: #FFFFFF;
  cursor: pointer;
  display: inline-block;
  font-family: "Proxima Nova", Arial, sans-serif;
  font-size: 1.2rem;
  font-weight: 600;
  line-height: 1;
  padding: 1em 2em;
  margin-bottom: 1em;
  text-decoration: none;
  transition: background-color 150ms ease;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap; }
  .Button--ghost {
    background-color: transparent;
    border: 1px solid #FFFFFF; }
    .Button--ghost:hover {
      background-color: rgba(255, 255, 255, 0.1); }

label {
  display: flex;
  flex-direction: column;
  align-items: flex-start; }

input {
  background-color: #FFFFFF;
  color: #636668;
  border: 1px solid #FFFFFF;
  border-radius: 3px;
  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.06);
  box-sizing: border-box;
  font-family: "Proxima Nova", Arial, sans-serif;
  font-size: 1.2rem;
  margin: 0 1em 1em 0;
  padding: 1em;
  transition: border-color 150ms ease; }

.WordFind {
  display: flex;
  flex-direction: column;
  margin: 2em; }
  .WordFind-config {
    display: flex;
    flex-direction: column;
    margin-top: 1em; }

.WordBank {
  width: 700px; }
  .WordBank-container {
    background-color: #FFFFFF;
    color: #636668;
    padding: 2em;
    display: flex;
    flex-wrap: wrap; }
  .WordBank-word {
    font-size: 1.6em;
    padding: 1em; }
  .WordBank-addContainer {
    display: flex;
    margin: 1em 0; }

.WordPuzzle {
  display: grid;
  justify-content: center;
  background-color: #FFFFFF;
  color: #636668;
  padding: 2em;
  margin: 1em;
  font-size: 2em; }
  .WordPuzzle--seven {
    grid-template-rows: repeat(7, auto);
    grid-template-columns: repeat(7, 40px); }
  .WordPuzzle-letter {
    padding: 0 10px; }
