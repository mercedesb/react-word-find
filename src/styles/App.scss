@import url("https://fonts.googleapis.com/css?family=Proxima+Nova");

$base-spacing: 10px;
$brand-color: #44957c;
$font-family: "Proxima Nova", Arial, sans-serif;
$white: #FFFFFF;
$black: #636668;
$max-num-columns: 15;

html {
  height: 100%;
}

body, #root {
  min-height:100%;
  display: flex;
  flex-direction: column;
  flex: 1;
  align-items: center;
  background-color: $brand-color;
}

.App {
  flex: 1;
  text-align: center;
  display: flex;
  flex-direction: column;
  background-color: $brand-color;
  font-family: $font-family;
  color: $white;
  min-height:100%;
  max-width: 1000px;
}

.Button {
  background-color: $brand-color;
  border: 0;
  border-radius: 3px;
  color: $white;
  cursor: pointer;
  display: inline-block;
  font-family: $font-family;
  font-size: 1.2rem;
  font-weight: 600;
  line-height: 1;
  padding: 1em 2em;
  margin-bottom: 1em;
  text-decoration: none;
  transition: background-color 150ms ease;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;

  &--ghost {
    background-color: transparent;
    border: 1px solid $white;

    &:hover {
      background-color: rgba(255, 255, 255, 0.1)
    }
  }
}

label {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

input {
  background-color: $white;
  color: $black;
  border: 1px solid $white;
  border-radius: 3px;
  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.06);
  box-sizing: border-box;
  font-family: $font-family;
  font-size: 1.2rem;
  margin: 0 1em 1em 0;
  padding: 1em;
  transition: border-color 150ms ease;
}

.WordFind {
  display: flex;
  flex-direction: column;
  margin: 2em;

  &-config {
    display: flex;
    flex-direction: column;
    margin-top: 1em;
  }
}

.WordBank {
  width: 700px;

  &-container {
    background-color: $white;
    color: $black;
    padding: 2em;
    display: flex;
    flex-wrap: wrap;
  }

  &-word {
    font-size: 1.6em;
    padding: 1em;
  }

  &-addContainer {
    display: flex;
    margin: 1em 0;
  }
}

.WordPuzzle {
  display: grid;
  justify-content: center;
  background-color: $white;
  color: $black;
  padding: 2em;
  margin: 1em;
  font-size: 2em;

  @for $i from 1 through $max-num-columns {
    &--#{$i} {
      grid-template-rows: repeat(#{$i}, auto);
      grid-template-columns: repeat(#{$i}, 40px);
    }
  }

  &-letter {
    padding: 0 $base-spacing;
  }
}