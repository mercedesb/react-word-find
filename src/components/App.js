import React, { Component } from 'react';
import '../styles/App.css';
import { WordBank } from './WordBank'
import { PuzzleGenerator } from './PuzzleGenerator'

class App extends Component {
  state = {
    words: ['hello', 'world'],
    shouldGenerate: false,
    size: 7
  }

  onAdd = (word) => {
    const currentWordBank = this.state.words
    currentWordBank.push(word)
    this.setState({words: currentWordBank})
  }

  onSizeChange = (event) => {
    this.setState({ size: event.target.value })
  }

  generateWordFind = () => {
    this.setState({ shouldGenerate: true })
  }

  buildPuzzle = () => {
    if (this.state.shouldGenerate) {
      return (
        <PuzzleGenerator
          wordBank={this.state.words}
        />
      )
    }
  }

  render () {
    return (
      <div className='App'>
        <div className='WordFind'>
          <WordBank
            words={this.state.words}
            onAdd={this.onAdd}
          />
          <div className='WordFind-config'>
            <label>
              Grid size
              <input 
                type='number'
                onChange={this.onSizeChange}
                value={this.state.size}
              />
            </label>
            <button className='Button Button--ghost' onClick={this.generateWordFind}>Generate Word Find</button>
          </div>
        </div>
        {
          this.buildPuzzle()
        } 
      </div>
    )
  }
}

export default App;

