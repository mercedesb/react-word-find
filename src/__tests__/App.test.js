import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import App from '../components/App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

let subject 

function loadSubject (props = {}) {
  subject = shallow(<App
    {...props}
  />)
}

describe('#render', () => {
  it('renders properly', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})

describe('#onAdd', () => {
  it('sets the state', () => {
    loadSubject()
    const word = 'newword'
    const initialState = subject.state('words')
    const expectedState = initialState.concat([word])
    subject.instance().onAdd(word)
    expect(subject.state('words')).toEqual(expectedState)
  })
})

describe('#onChange', () => {
  it('sets the state', () => {
    loadSubject()
    subject.instance().onSizeChange({ target: { value: 8 } })
    expect(subject.state('size')).toEqual(8)
  })
})

describe('#generateWordFind', () => {
  it('sets the state', () => {
    loadSubject()
    subject.instance().generateWordFind()
    expect(subject.state('shouldGenerate')).toEqual(true)
  })
})