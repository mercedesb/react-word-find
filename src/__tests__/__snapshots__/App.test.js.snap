// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WordBank
          onAdd={[Function]}
          words={
                    Array [
                              "hello",
                              "world",
                            ]
          }
/>,
        <button
          className="Button Button--ghost"
          onClick={[Function]}
>
          Generate Word Find
</button>,
        undefined,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onAdd": [Function],
          "words": Array [
            "hello",
            "world",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Generate Word Find",
          "className": "Button Button--ghost",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Generate Word Find",
        "type": "button",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WordBank
            onAdd={[Function]}
            words={
                        Array [
                                    "hello",
                                    "world",
                                  ]
            }
/>,
          <button
            className="Button Button--ghost"
            onClick={[Function]}
>
            Generate Word Find
</button>,
          undefined,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onAdd": [Function],
            "words": Array [
              "hello",
              "world",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Generate Word Find",
            "className": "Button Button--ghost",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Generate Word Find",
          "type": "button",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
